### DISCUSSION TITLE
Why did `iced` trigger events like `on_clicked`  at `ButtonPressed` rather than `ButtonReleased`

### DISCUSSION THREAD

**Discussion by wayhoowang** [hecrj/iced#2045] [Q&A] [answered]
I believe the latter one is a more common choice.

**Comment by hecrj** [comment]
There is no occurence of `on_clicked` in the public API.

**Reply by wayhoowang** [reply]
Sorry that I didn't make my question completely clear. I will try to give a more concrete example to explain what I am concerning about. The Example is:
```
        let pick_list = pick_list(
            &Language::ALL[..],
            self.selected_language,
            Message::LanguageSelected,
        )
```
Then, `Message::LanguageSelected` will be sent to `update` once I pressed the button on my mouse. However, I believe in most UI frameworks, the event won't be sent to update logic until I release the button. I am wondering why did `iced` adopted an non-mainstream choice, and if there is any chances that I can customize it without hacking into `iced`.

**Reply by hecrj** [reply, marked as answer]
There is no particular reason. Just a mishap.
